src/Sylius/Bundle/CoreBundle/Tests/Form/Type/CatalogPromotion/CatalogPromotionActionTypeTest.php

```diff
@@ -0,0 +1,307 @@
+    {
+        $fixedDiscount = $this->setupFixedDiscount();
+
+        $this->channelRepository->findAll(Argument::any())->willReturn(
+            [$this->channel->reveal()]
+        );
+
+        $form = $this->factory->create(CatalogPromotionActionType::class, $fixedDiscount);
+        $form->submit([
+            'type' => 'fixed_discount',
+            'configuration' => [
+                'WEB_US' => [
+                    'amount' => 20
+                ]
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertTrue($form->isSynchronized());
+        $this->assertSame('fixed_discount', $catalogPromotionAction->getType());
+        $this->assertSame(['WEB_US' => ['amount' => 20]], $catalogPromotionAction->getConfiguration());
+    }
+
+    /** @test */
+    public function it_updates_amount_of_percentage_discount(): void
+    {
+        $percentageDiscount = $this->setupPercentageDiscount();
+
+        $this->channelRepository->findAll(Argument::any())->willReturn(
+            [$this->channel->reveal()]
+        );
+
+        $form = $this->factory->create(CatalogPromotionActionType::class, $percentageDiscount);
+        $form->submit([
+            'type' => 'percentage_discount',
+            'configuration' => [
+                'amount' => 10
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertSame('percentage_discount', $catalogPromotionAction->getType());
+        $this->assertSame(['amount' => 0.1], $catalogPromotionAction->getConfiguration());
+    }
+
+    /** @test */
+    public function it_changes_action_to_fixed_discount(): void
+    {
+        $percentageDiscount = $this->setupPercentageDiscount();
+
+        $this->channelRepository->findAll(Argument::any())->willReturn(
+            [$this->channel->reveal()]
+        );
+
+        $form = $this->factory->create(CatalogPromotionActionType::class, $percentageDiscount);
+        $form->submit([
+            'type' => 'fixed_discount',
+            'configuration' => [
+                'WEB_US' => [
+                    'amount' => 10
+                ]
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertSame('fixed_discount', $catalogPromotionAction->getType());
+        $this->assertSame(['WEB_US' => ['amount' => 10]], $catalogPromotionAction->getConfiguration());
+        $this->assertSame(['WEB_US' => ['amount' => 10]], $catalogPromotionAction->getConfiguration());
```

Comment:
Did you want to check something else in the second line? 
```suggestion
        $this->assertSame(['WEB_US' => ['amount' => 10]], $catalogPromotionAction->getConfiguration());
```